### PR TITLE
ui: add recent statements to sql activity page

### PR DESCRIPTION
**Pull request by amyyq2**
Previously, we only displayed Active Statements on the SQL Activity Page. This change adds Recent Statements to the previous Active Executions table. This change also fixes pagination on that table, and adds four new
possible Execution Statuses: Failed, Timed Out,
Canceled, and Completed, along with a corresponding icon color.

Video with changes: https://www.loom.com/share/06fe2e67bc8f41178f6f7c140e263e9e
The video shows the UI after running several queries via the CLI.

Part of #86955

Release note (ui change): adds Recent Statements to the SQL Activity page

### CODE REVIEW

**Review by maryliag — COMMENTED** [review]
Nice!
Can you add a video/images with the changes? 

> Reviewed 5 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-:-NHQdkYq6YUnOVtK5zTQ:b-hx8tsb)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Review by xinhaoz — COMMENTED** [review]
Looking good! +1 to Marylia's comment on images and videos though.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-:-NHQeT1uBCwabs-RNVoE:b6shscw)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @amyyq2 and @maryliag)

___
*[`pkg/ui/workspaces/cluster-ui/src/recentExecutions/recentStatementUtils.ts` line 134 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-NHQd3Fi48aCf3Z6SgbD:-NHQd3Fi48aCf3Z6SgbE:b185r4) ([raw file](https://github.com/cockroachdb/cockroach/blob/98826104164ece8ecd8bf3de669c9b7fa5b6e93e/pkg/ui/workspaces/cluster-ui/src/recentExecutions/recentStatementUtils.ts#L134)):*
> ```XML
>         const queryTxnID = byteArrayToUuid(query.txn_id);
>         let status: ExecutionStatus;
>         switch (query.phase) {
> ```

Can you lift this out into its own function `queryPhaseToExecutionStatus` or something and also use it for the active stmts?



_Code quote:_
```XML
          case ActiveStatementPhase.EXECUTING:
            status = "Executing";
            break;
          case ActiveStatementPhase.PREPARING:
            status = "Preparing";
            break;
          case ActiveStatementPhase.CANCELED:
            status = "Canceled";
            break;
          case ActiveStatementPhase.TIMED_OUT:
            status = "Timed Out";
            break;
          case ActiveStatementPhase.COMPLETED:
            status = "Completed";
            break;
          case ActiveStatementPhase.FAILED:
            status = "Failed";
            break;
```
___
*[`pkg/ui/workspaces/cluster-ui/src/recentExecutions/statusIcon.tsx` line 24 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-NHQdY0KEFF8LFd_4mXP:-NHQdY0KEFF8LFd_4mXQ:b-i3w1r0) ([raw file](https://github.com/cockroachdb/cockroach/blob/98826104164ece8ecd8bf3de669c9b7fa5b6e93e/pkg/ui/workspaces/cluster-ui/src/recentExecutions/statusIcon.tsx#L24)):*
> ```XML
> 
> export const StatusIcon: React.FC<StatusIconProps> = ({ status }) => {
>   const statusClassName = status.toLowerCase().replace(/\s/g, "_");
> ```

Can you comment what the class name will look like for clarity? Something like Class names will be execution statuses with spaces converted to underscores, e.g. `Timed Out` -> `timed_out`.

**Review by amyyq2 — COMMENTED** [review]
Thank you, and ok I've added a video!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-:-NHQiMp-0oAFsPpj8JS0:b-coh0bv)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @maryliag and @xinhaoz)

**Review by amyyq2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92154)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @maryliag and @xinhaoz)

___
*[`pkg/ui/workspaces/cluster-ui/src/recentExecutions/recentStatementUtils.ts` line 134 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-NHQd3Fi48aCf3Z6SgbD:-NHSVLmS1WlZz-Kg53V_:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/98826104164ece8ecd8bf3de669c9b7fa5b6e93e/pkg/ui/workspaces/cluster-ui/src/recentExecutions/recentStatementUtils.ts#L134)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Can you lift this out into its own function `queryPhaseToExecutionStatus` or something and also use it for the active stmts?
</blockquote></details>

Done.
___
*[`pkg/ui/workspaces/cluster-ui/src/recentExecutions/statusIcon.tsx` line 24 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-NHQdY0KEFF8LFd_4mXP:-NHSVMRc6UUfgf1tmb9w:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/98826104164ece8ecd8bf3de669c9b7fa5b6e93e/pkg/ui/workspaces/cluster-ui/src/recentExecutions/statusIcon.tsx#L24)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Can you comment what the class name will look like for clarity? Something like Class names will be execution statuses with spaces converted to underscores, e.g. `Timed Out` -> `timed_out`.
</blockquote></details>

Done.

**Review by xinhaoz — APPROVED** [review]
:lgtm: Nice!

> Reviewed 1 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-:-NHV1rGQE76mxirFumyT:b-8gk8uj)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @maryliag)

**Review by maryliag — APPROVED** [review]
:lgtm:  (once backend PR gets merged)
Awesome work on this!

> Reviewed 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-:-NHkgphX83CzX_R1fl_Z:b66p4mm)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @xinhaoz)

**Review by j82w — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92154)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @amyyq2 and @xinhaoz)

___
*[`-- commits` line 4 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-NHyDl856G6PBKAHigRB):*
What is the benefit of keeping a separate tab for recent statements instead of just merging them into a single table with the date time picker?

**Review by amyyq2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92154)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @j82w and @xinhaoz)

___
*[`-- commits` line 4 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-NHyDl856G6PBKAHigRB):*
<details><summary><i>Previously, j82w wrote…</i></summary><blockquote>

What is the benefit of keeping a separate tab for recent statements instead of just merging them into a single table with the date time picker?
</blockquote></details>

With this change, I'm adding recent statements to the existing tab for "Active Statements". I renamed the table and tab to be "Recent Statements", so it will contain both active and recently finished statements. 

Is the previous Active Statements table the single table with the date time picker you're referring to, or is there something else I can merge recent statements into?

**Comment by j82w** [comment]
*[`-- commits` line 4 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/92154#-NHyDl856G6PBKAHigRB):*
<details><summary><i>Previously, amyyq2 wrote…</i></summary><blockquote>

With this change, I'm adding recent statements to the existing tab for "Active Statements". I renamed the table and tab to be "Recent Statements", so it will contain both active and recently finished statements. 

Is the previous Active Statements table the single table with the date time picker you're referring to, or is there something else I can merge recent statements into?
</blockquote></details>

This changes looks good to me. I'm wondering if it's possible to further simplify the user experience. @kevin-v-ngo is there a reason there is a separate tab for recent statements instead of just having one page with a filter option?

**Comment by amyyq2** [comment]
Closing this PR, as it was part of an old design. New PR is here: https://github.com/cockroachdb/cockroach/pull/94159

**Pull request closed without merging**
